test/integration/suite_test.go

```diff
@@ -245,6 +249,12 @@ func TestMain(m *testing.M) {
 		_ = env.Cluster().Client().NetworkingV1().IngressClasses().Delete(ctx, iclass.Name, metav1.DeleteOptions{})
 	}()
 
+	if os.Getenv("TEST_RUN_INVALID_CONFIG_CASES") == "true" {
```

Comment:
Are we going to run these tests in CI? If not, I'd say that we shouldn't keep them as it's very likely they will rot when not run regularly. Given we have the E2E covering this case, does the integration test check anything more?

Reply:
OK, will add this case to CI. This is not covered by e2e case currently. IMO it is better to include the cases in both integration tests and e2e tests, because they are actually in different environments.

Reply:
added one more step to nightly to run the cases with invalid config.